### PR TITLE
[processor/interval] Update config structure for interval processor

### PR DESCRIPTION
**Pull request by ArthurSens**
**Description:** This PR addresses comments made in #34805 after it was merged

The gist is change in configuration from

```yaml
interval: 60s
gauge_pass_through: false
summary_pass_through: false
```

to 

```yaml
interval: 60s
pass_through:
  gauge: false
  summary: false
```

**Link to tracking Issue:** #34920

### CODE REVIEW

**Comment by ArthurSens** [comment]
If we get this PR merged before the next release (I don't know when that will happen), we won't introduce any breaking changes. It also won't make sense to add a changelog entry in this case.

If a release is made, then we'll introduce breaking changes 😬

**Review by RichieSams — APPROVED** [review]

File: processor/intervalprocessor/README.md
Comment (sh0rez) [inline processor/intervalprocessor/README.md:41]:
```suggestion
intervalprocessor:
  # The interval in which the processor should export the aggregated metrics. 
  [ interval: <duration> | default = 60s ]
  
  pass_through:
    # Whether gauges should be aggregated or passed through to the next component as they are
    [ gauge: <bool> | default = false ]
    # Whether summaries should be aggregated or passed through to the next component as they are
    [ summary: <boo>l | default = false ]
```

File: processor/intervalprocessor/processor.go
Comment (sh0rez) [inline processor/intervalprocessor/processor.go:69]: how about just storing the entire `config` struct in `processor`, instead of manually copying literally all fields?

**Review by tombrk — APPROVED** [review]
minor nits, LGTM

File: processor/intervalprocessor/processor.go
Comment (ArthurSens) [reply processor/intervalprocessor/processor.go:69]: Good idea

**Comment by jpkrohling** [comment]
Marking as skip changelog, as this doesn't introduce user-facing changes, this is part of the initial available version.

**Review by MovieStoreGuy — APPROVED** [review]

**Review by tombrk — APPROVED** [review]

**Review by RichieSams — APPROVED** [review]

**Comment by ArthurSens** [comment]
Asking out of curiosity... should this PR have the label "ready-to-merge"?

**Comment by MovieStoreGuy** [comment]
> Asking out of curiosity... should this PR have the label "ready-to-merge"?

It is a manual step for approvers to add if they are waiting for a maintainer to merge.

**Pull request merged**
